weavelet.go

```diff
@@ -60,8 +60,15 @@ type weavelet struct {
 	componentsByName map[string]*component       // component name -> component
 	componentsByType map[reflect.Type]*component // component type -> component
 
+	// TODO(mwhittaker): We have one client for every component. Every client
```

Comment:
True that!

Reply:
I think I have a not-too-ugly way of fixing this that I'll do in a future PR.